apm-agent-plugins/apm-httpclient11-plugin/src/test/java/co/elastic/apm/agent/httpclient/HttpClientAsyncInstrumentationTest.java

```diff
@@ -0,0 +1,62 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 - 2020 Elastic and contributors
+ * %%
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.httpclient;
+
+import org.junit.Before;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.concurrent.CompletableFuture;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HttpClientAsyncInstrumentationTest extends AbstractHttpClientInstrumentationTest {
+    private HttpClient client;
+
+    @Before
+    public void setUp() {
+        client = HttpClient.newBuilder().followRedirects(HttpClient.Redirect.NORMAL).build();
+    }
+
+    @Override
+    protected void performGet(String path) throws Exception {
+        HttpRequest request = HttpRequest.newBuilder()
+            .uri(URI.create(path))
+            .build();
+        client.sendAsync(request, HttpResponse.BodyHandlers.ofString()).get();
+    }
+
+    @Override
+    protected boolean isIpv6Supported() {
+        return true;
+    }
+
+    @Override
+    public void assertCircularRedirect() {
+        assertThat(reporter.getSpans().get(0).getContext().getHttp().getStatusCode()).isEqualTo(303);
```

Comment:
Why is this necessary? Doesn't this fail on circular redirects? If so, override the test method to just skip it.

Reply:
Yes, this fail on circluar redirects.
Ok, I overrided this test method.